x/mock/simulation/rand_util.go

```diff
@@ -37,14 +38,37 @@ func RandStringOfLength(r *rand.Rand, n int) string {
 }
 
 // Generate a random amount
+// Note: The range of RandomAmount includes max, and is, in fact, biased to return max.
 func RandomAmount(r *rand.Rand, max sdk.Int) sdk.Int {
-	return sdk.NewInt(int64(r.Intn(int(max.Int64()))))
+	// return sdk.NewInt(int64(r.Intn(int(max.Int64()))))
+	max2 := big.NewInt(0).Mul(max.BigInt(), big.NewInt(2))
+	randInt := big.NewInt(0).Rand(r, max2)
+	if randInt.Cmp(max.BigInt()) > 0 {
+		randInt = max.BigInt()
+	}
+	result := sdk.NewIntFromBigInt(randInt)
+	// Sanity
+	if result.GT(max) {
+		panic(fmt.Sprintf("%v > %v", result, max))
+	}
+	return result
 }
 
 // RandomDecAmount generates a random decimal amount
+// Note: The range of RandomDecAmount includes max, and is, in fact, biased to return max.
 func RandomDecAmount(r *rand.Rand, max sdk.Dec) sdk.Dec {
-	randInt := big.NewInt(0).Rand(r, max.Int)
-	return sdk.NewDecFromBigIntWithPrec(randInt, sdk.Precision)
+	// randInt := big.NewInt(0).Rand(r, max.Int)
```

Comment:
Make this call RandomAmount so that we can tweak that one function...
in the future we could make RandomAmount's probability to include bias for 0 and small numbers as well.

Reply:
i.e. keeps the code dry in a good way.

Reply:
Do we actually want to keep said changes as it's biased?